src/rdf4cpp/rdf/parser/RDFFileParser.hpp

```diff
@@ -34,14 +34,14 @@ class RDFFileParser {
     explicit RDFFileParser(std::string &&file_path, ParsingFlags flags = ParsingFlags::none(),
                            storage::node::NodeStorage node_storage = storage::node::NodeStorage::default_instance());
 
-    class Iterator {
+    class iterator {
         friend class RDFFileParser;
-        friend bool operator==(const RDFFileParser::Iterator &iter, std::default_sentinel_t) noexcept;
-        std::unique_ptr<std::ifstream> stream_;
+        friend bool operator==(const RDFFileParser::iterator &iter, std::default_sentinel_t) noexcept;
+        std::unique_ptr<FILE, decltype([](FILE *f) { fclose(f); })> stream_;
```

Comment:
`FILE` should actually not be available here but only `std::FILE`. Maybe this is what confuses gcc?
- https://en.cppreference.com/w/cpp/header/cstdio
- https://en.cppreference.com/w/cpp/io/c/FILE

Reply:
No, thats not it. (And yes technically only std::FILE should be available but libstdc++ exposes everything from the c headers in the global namespace as well. Probably because they don't want to annoy people)

Reply:
Removed the unique_ptr, gcc seems to not like in in combination with `FILE`